src/pilot/assign_role.py

```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
     return raid_controller_ids
 
 
-def define_controller_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
+    all_physical_disks = drac_client.list_physical_disks()
 
-    raid_10_logical_disk = define_single_raid_10_logical_disk(
-        drac_client, raid_controller_ids[0])
+    # Get the drives controlled by the RAID controller
+    raid_cntlr_physical_disks = {}
+    for disk in all_physical_disks:
+        if disk.controller in raid_controllers:
+            if disk.controller in raid_cntlr_physical_disks:
+                raid_cntlr_physical_disks[disk.controller].append(disk)
+            else:
+                raid_cntlr_physical_disks[disk.controller] = [disk]
+
+    boss_controller = _fetch_boss_controller(drac_client, raid_controllers)
+    if boss_controller:
+        if check_cntlr_physical_disks_len(
+           raid_cntlr_physical_disks[boss_controller]):
+            os_logical_disk = define_storage_operating_system_logical_disk(
```

Comment:
Having this function, which is used for the control and compute roles, call another function which is named for the storage role can be confusing.

Let us make the following function name changes:

1. `define_storage_operating_system_logical_disk` to `define_operating_system_logical_disk`
2. `find_physical_disks_for_storage_os` to `find_physical_disks_for_os`

Reply:
Done